test/com/google/javascript/jscomp/VarCheckTest.java

```diff
@@ -173,6 +173,7 @@ public void testNamedClass() {
     testSameEs6("class x {}");
     testSameEs6("var x = class x {};");
     testSameEs6("var y = class x {};");
+    testSameEs6("var y = class x { foo() { return new x; } };");
```

Comment:
Let's add a test to demonstrate that the class expression name is in scope inside the extends clause.